examples/amqp/src/main/java/io/cloudevents/examples/amqp/vertx/AmqpClient.java

```diff
@@ -0,0 +1,160 @@
+
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.qpid.proton.amqp.messaging.Accepted;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.message.Message;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.amqp.ProtonAmqpMessageFactory;
+import io.cloudevents.core.message.MessageReader;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonClient;
+import io.vertx.proton.ProtonClientOptions;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonHelper;
+import io.vertx.proton.ProtonMessageHandler;
+import io.vertx.proton.ProtonQoS;
+import io.vertx.proton.ProtonReceiver;
+import io.vertx.proton.ProtonSender;
+
+/**
+ * A example vertx-based AMQP client that interacts with a remote AMQP server to send and receive CloudEvent messages.
+ */
+public class AmqpClient {
+
+    private static ProtonConnection connection;
+
+    private static final String SERVER_HOST = "127.0.0.1";
+    private static final int SERVER_PORT = 5672;
+    private static final String SEND_MESSAGE = "send";
+    private static final String RECEIVE_MESSAGE = "receive";
+
+    private static PrintWriter writer = new PrintWriter(System.out, true);
+
+    public static void main(String args[]) {
+
+        if (args.length < 1) {
+            writer.println("Usage: AmqpClient [send|receive]");
+            return;
+        }
+
+        final String action = args[0].toLowerCase();
+
+        final Vertx vertx = Vertx.vertx();
+
+        switch (action) {
+
+        case SEND_MESSAGE:
+            connectToServer(vertx, SERVER_HOST, SERVER_PORT)
+                    .compose(conn -> {
+                        connection = conn;
+                        writer.printf("[Client] Connected to %s:%s", SERVER_HOST, SERVER_PORT);
+
+                        return openSenderLink();
+                    }).map(sender -> {
+
+                        final JsonObject payload = new JsonObject().put("temp", 50);
+                        final String to = "/telemetry";
+
+                        final Message message = ProtonHelper.message(to, payload.toString());
+
+                        // set attributes
+                        final Map<String, Object> attributes = new HashMap<>();
+                        attributes.put("cloudEvents:type", "com.example.sampletype1");
+                        attributes.put("cloudEvents:source", "http://127.0.0.1/amqp-client");
+                        attributes.put("cloudEvents:id", "client-id");
+                        attributes.put("cloudEvents:specversion", "1.0");
+                        attributes.put("cloudEvents:time", "2020-11-06T21:47:12.037467+00:00");
+                        message.setApplicationProperties(new ApplicationProperties(attributes));
+
+                        sender.send(message, delivery -> {
```

Comment:
Why are you manually writing the event here?

Reply:
i dont understand. can you please explain what you mean?

i am sending a proton amqp message to the server and adding  the cloudevent attribute names and values as application-properties.

Reply:
Could you create a `CloudEvent` and write it using the message writer more than manually writing the attributes here?